### ISSUE TITLE
Which framework to use for more resource-efficient training? Plus a recent paper in this regard, "FastSiam: Resource-Efficient Self-supervised Learning on a Single GPU"

### ISSUE THREAD

**Issue by faris-k** [lightly-ai/lightly#972]
For those of us with just a single GPU, what framework do you suggest using in general? I know that SimSiam is often a good choice in this regard since it generally requires smaller batch sizes than the other frameworks.

I recently came across the following paper, "FastSiam: Resource-Efficient Self-supervised Learning on a Single GPU": [https://link.springer.com/chapter/10.1007/978-3-031-16788-1_4](https://link.springer.com/chapter/10.1007/978-3-031-16788-1_4). I'm not well-versed in how to judge how good all these new self-supervised frameworks are, but it seems promising to have a framework like this that is targeted to those in the community who don't have access to big compute power. No code implementations of this anywhere 😞 Are there any plans for Lightly to include frameworks like these, which aim to be "lighter" as opposed to "more powerful"?

![image](https://user-images.githubusercontent.com/68213622/199854357-21ac4bf5-c7c9-4ae5-8f8c-ee8fd575dd4c.png)
![image](https://user-images.githubusercontent.com/68213622/199854173-103889fc-c005-4348-9307-8340c95c556f.png)
![image](https://user-images.githubusercontent.com/68213622/199854144-595da144-e8d5-41c3-9a9f-76b91af59ac5.png)

**Comment by philippmwirth** [comment]
Hi @faris-k,

In our experience, both SimCLR and MoCo work well even with smaller batchsizes whereas SimSiam can be a bit tricky to train.

Regarding FastSiam: We'll have to take a closer look at the paper in one of our paper sessions (they're regularly organized in our [Discord](https://discord.gg/xvNJW94)). At first sight it looks like it could already be implemented with the building blocks currently available in the package.

I'll keep you posted about any updates 🙂

**Comment by faris-k** [comment]
For a class project, I implemented FastSiam just by modifying SimSiam and using `MultiViewCollateFunction`. I had pretty good results with it (albeit this was just for our domain-specific problem). We tried benchmarking it on Imagenette like all the other models in lightly, and it performed comparably with SimSiam and SimCLR (we used consumer-grade hardware though, so nothing fancy and we can't really say how it compares speed-wise with some of the other techniques). Just wanted to chime in with an update 🙂 It's nice to have packages like lightly that are easy to use and fairly plug-and-play 😊

Implementation here: https://github.com/faris-k/fastsiam-wafers
